Explain about SystemPackageTool and sudo (#1127)
conan-io/docs: reference/conanfile/methods.rst

The use of ``sudo`` in the internals of the ``install()`` and ``update()`` methods is controlled by the ``CONAN_SYSREQUIRES_SUDO``
environment variable, so if the users don't need sudo permissions, it is easy to opt-in/out.

> When ``CONAN_SYSREQUIRES_SUDO`` is not defined, Conan will mimic if it is possible to use ``sudo`` based on:

Reviewer: ```suggestion
When the environemtn variable ``CONAN_SYSREQUIRES_SUDO`` is not defined, Conan will try to use :command:`sudo` if the following conditions are met:
```

>     - Is ``sudo`` available in ``PATH``

Reviewer: ```suggestion
    - :command:`sudo` is available in the ``PATH``.
```

>     - The platform name should be ``posix`` and the UID (user id) should not be ``0``

Reviewer: ```suggestion
    - The platform name is ``posix`` and the UID (user id) is not ``0``
```

The use of ``sudo`` in the internals of the ``install()`` and ``update()`` methods is controlled by the ``CONAN_SYSREQUIRES_SUDO``
environment variable, so if the users don't need sudo permissions, it is easy to opt-in/out.

When ``CONAN_SYSREQUIRES_SUDO`` is not defined, Conan will mimic if it is possible to use ``sudo`` based on:

    - Is ``sudo`` available in ``PATH``
    - The platform name should be ``posix``

>     - UID (user id) should be ``0``

Reviewer 2: I dont think so, the code is:
```python
if os.name == 'posix' and os.geteuid() == 0:
                return False
```
Which means it wont use sudo if posix and geteuid =0

Author: good catch!

I gonna change to:

`- The platform name should be `posix` and the UID (user id) should not be `0` `

Author: Done! Please, comment again.
